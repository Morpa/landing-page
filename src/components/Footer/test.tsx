import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the names of authors', () => {
    const { container } = renderWithTheme(<Footer />)

    expect(screen.getByText('Willian Justen').closest('a')).toHaveAttribute(
      'href',
      'https://willianjusten.com.br/'
    )

    expect(screen.getByText('Guilherme Louro').closest('a')).toHaveAttribute(
      'href',
      'https://guilouro.dev'
    )

    expect(screen.getByText('Marcos Oliveira').closest('a')).toHaveAttribute(
      'href',
      'https://dribbble.com/vmarcosp'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
