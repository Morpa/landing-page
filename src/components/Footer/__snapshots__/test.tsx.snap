// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the names of authors 1`] = `
.c1 {
  max-width: 130rem;
  padding: 6.0rem calc(3.2rem / 2);
  margin: 0 auto;
}

.c0 {
  background: #FFFFFF;
  color: #7B7373;
  text-align: center;
  font-size: 1.3rem;
  line-height: 3.4rem;
  position: relative;
  z-index: 2;
  margin-top: -3.2rem;
}

.c0 a {
  color: #F231A5;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid #F231A5;
}

@media (min-width:768px) {
  .c1 {
    padding: 6.0rem calc(3.2rem / 2);
  }
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.8rem;
    margin-top: -12.8rem;
    padding-bottom: 6.0rem;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <p>
      Desenvolvido por
      <a
        href="https://willianjusten.com.br/"
      >
        Willian Justen 
      </a>
      e
      <a
        href="https://guilouro.dev"
      >
         Guilherme Louro
      </a>
    </p>
    <p>
      Design por 
      <a
        href="https://dribbble.com/vmarcosp"
      >
        Marcos Oliveira
      </a>
    </p>
  </div>
</footer>
`;
